crates/static-analysis-kernel/src/analysis/javascript.rs

```diff
@@ -218,6 +134,12 @@ fn execute_rule_internal(
     // node context with the file-context we calculated for each file.
     let js_code = format!(
         r#"
+(() => {{
+// Clear arrays that might've been populated by a prior node's execution.
+// NOTE: These variables exist in the scope immediately outside of this IIFE.
+stellaAllErrors.length = 0;
+console.lines.length = 0;
```

Comment:
1. Why not reassign an empty array?
2. If we want to have a function that cleans the runtime, we should put this in `stella.js` and call it here.

More globally, I wonder why we need to do this. I believe that it's not necessary since we run an[ isolate](https://deno.com/blog/anatomy-isolate-cloud) for each execution. So the environment and variables should not be shared?

Reply:
Agreed. Let's have a function in `stella.js` that clean the state instead of putting it here.